src/components/ChangesNotAppliedModal.jsx

```diff
@@ -1,37 +1,51 @@
 import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import {
   Modal, Typography, Space, Button,
 } from 'antd';
+import { useRouter } from 'next/router';
+import { discardChangedQCFilters, navigateFromProcessingTo } from '../redux/actions/experimentSettings';
+import { runPipeline } from '../redux/actions/pipeline';
 
 import { getUserFriendlyQCStepName } from '../utils/qcSteps';
 
 const { Text } = Typography;
 
 const ChangesNotAppliedModal = (props) => {
-  const {
-    steps, visible, onRun, onDiscard, onCancel,
-  } = props;
+  const { experimentId } = props;
+
+  const router = useRouter();
+  const dispatch = useDispatch();
+
+  const { changedQCFilters, navigationPath } = useSelector((state) => state.experimentSettings.processing.meta);
 
   return (
     <Modal
-      visible={visible}
+      visible={navigationPath}
```

Comment:
Why navigation path is being used in visible?

Reply:
It's a shortcut basically, if it's an empty string the modal doesn't show.

Reply:
We should probably use null instead of an empty string, and perhaps to make it more explicit use `navigationPath === null`